src/xwayland/xwm.c

```diff
@@ -221,6 +221,7 @@ read_properties(struct wlc_xwm *xwm, struct wlc_x11_window *win, const xcb_atom_
          wlc_dlog(WLC_DBG_XWM, "WM_TRANSIENT_FOR: %u", *xid);
       } else if (props[i] == x11.atoms[NET_WM_PID] && reply->type == XCB_ATOM_CARDINAL) {
          // PID
+         wlc_view_set_pid(view, *(pid_t *)xcb_get_property_value(reply));
```

Comment:
Seems to identical to weston. Weston seems to do some validation on the pid though, not sure if that's neccessary. Not sure about the inner workings of xcb_get_property_value, but I hope it returns data with sizeof(1) since otherwise this could cause aliasing issues and thus memcpy would be correct way to get the data. But since weston does it the same way I assume it's ok.